src/ValidationFailure.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\XmlValidator;
+
+final class ValidationFailure
+{
+    private $message;
+    private $line;
+
+    public function __construct(string $message, int $line)
+    {
+        $this->message = $message;
+        $this->line = $line;
+    }
+
+    public function getMessage() : string
+    {
+        return $this->message;
+    }
+
+    public function getLine() : int
```

Comment:
Some cases may not have a line number (eg invalid schema), maybe should make it optional, or have separate classes for line and file failures?

Reply:
Separate classes would still need a base interface for ease of writing `catch` statements, so I'd go for an optional value